design-documents/graph-ql/coverage/customer-orders.md

```diff
@@ -0,0 +1,210 @@
+# Customer Orders API
+
+## Overview
+
+The GraphQL API should provide a possibility to retrieve orders, shipments, invoices, credit memos for the logged in customer. The current schema allows fetching only simple order details and doesn't provide a possibility to fetch order details by a number.
+
+The proposed queries might look like:
+
+```graphql
+# query to return a list of all customer orders
+type Query {
+    customerOrders (
+        currentPage: Int = 1 # current page of the customer order list. default is 1.
+        pageSize: Int = 20 # page size for the customer orders list. default is 20.
+    ): CustomerOrders
+}
+
+# collection of customer orders that contains pagination and individual order details.
+type CustomerOrders {
+    items: [CustomerOrder] # collection of customer orders that contains individual order details.
+    currentPage: Int # current page of the customer order list. default is 1.
+    pageSize: Int # page size for the customer orders list. default is 20.
+}
+
+# query to return an individual order detail by order number.
+type Query {
+    customerOrder (
+        order_number: String # order number for the query input.
+    ): CustomerOrder   
+}
+```
+## Order Type Schema
+
+The proposed type for the customer order might look like:
+
+```graphql
+# order details
+type CustomerOrder {
+    order_date: String! # date when the order was placed.
+    status: String! # current status of the order.
+    order_number: String! # order number.
+    items: [OrderProductInterface] # collection of all the items purchased
+    order_pricing: OrderPricingInterface! # pricing details for the order.
+    invoice_details: [Invoice] # invoice details for the order.
+    shipping_details: [OrderShipment] # shipping details for the order.
+    payment_method: [PaymentMethod] # payment details for the order.
+}
+```
+
+The `grand_total`, `id`, `increment_id`, `created_at` will be marked as deprecated.
+
+### Order Item
+
+The order items will be presented as separate interface which will have multiple implementations for invoice, shipment and credit memo types.
+
+```graphql
+interface OrderProductInterface {
+    product_name: String # name of the base product.
+    sku: String! # sku of the base product.
+    url: String # url of the base product.
+    child_products: [ChildOrderProductInterface] # child products in case of bundled products.
```

Comment:
Is bundle the only case? What about configurable?

Reply:
Configurable products are presented by simple with options. Each option is a variation of a simple product.